### PR TITLE
import export setup

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

File: pyaedt/modules/SolveSetup.py [hunk 1]
```diff
@@ -1499,6 +1499,36 @@ def add_sweep(self, sweepname=None, sweeptype="Interpolating"):
...
 
+    def import_from_json(self, fpath):
+        """Import setup properties from a json file.
+
+        Parameters
+        ----------
+        fpath : str
+            File path of the json file.
+        """
+        self.props._import_properties_from_json(fpath)
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:1510]:
```suggestion
    def import_from_json(self, file_path):
        """Import setup properties from a json file.

        Parameters
        ----------
        file_path : str
            File path of the json file.
        """
        self.props._import_properties_from_json(file_path)
```

File: pyaedt/modules/SolveSetup.py [hunk 2]
```diff
@@ -1499,6 +1499,36 @@ def add_sweep(self, sweepname=None, sweeptype="Interpolating"):
...
+
+    def export_to_json(self, fpath):
+        """Export all setup properties into a json file.
+
+        Parameters
+        ----------
+        fpath : str
+            File path of the json file.
+        """
+        return self.props._export_properties_to_json(fpath)
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:1530]:
```suggestion
    def export_to_json(self, file_path):
        """Export all setup properties into a json file.

        Parameters
        ----------
        file_path : str
            File path of the json file.
        """
        return self.props._export_properties_to_json(file_path)
```

File: pyaedt/modules/SetupTemplates.py [hunk 3]
```diff
@@ -2172,3 +2173,40 @@ def __init__(self, setup, props):
...
+        def set_props(target, source):
+            for k, v in source.items():
+                if k not in target:
+                    raise Exception("{} is not a valid property name.".format(k))
+                if not isinstance(v, dict):
+                    target[k] = v
+                else:
+                    set_props(target[k], v)
+
+        with open(fpath, "r", encoding="utf-8") as f:
```
Comment (MaxJPRey) [inline pyaedt/modules/SetupTemplates.py:2209]:
```suggestion
        with open(file_path, "r", encoding="utf-8") as f:
```

File: pyaedt/modules/SetupTemplates.py [hunk 4]
```diff
@@ -2172,3 +2173,40 @@ def __init__(self, setup, props):
...
+        with open(fpath, "w", encoding="utf-8") as f:
+            f.write(json.dumps(self, indent=4, ensure_ascii=False))
+        return True
+
+    def _import_properties_from_json(self, fpath):
+        """Import setup properties from a json file.
+
+        Parameters
+        ----------
+        fpath : str
```
Comment (MaxJPRey) [inline pyaedt/modules/SetupTemplates.py:2196]:
```suggestion
        file_path : str
```

File: pyaedt/modules/SetupTemplates.py [hunk 5]
```diff
@@ -2172,3 +2173,40 @@ def __init__(self, setup, props):
...
+        """Export all setup properties into a json file.
+
+        Parameters
+        ----------
+        fpath : str
+            File path of the json file.
+        """
+        if not fpath.endswith(".json"):
+            fpath = fpath + ".json"
+        with open(fpath, "w", encoding="utf-8") as f:
```
Comment (MaxJPRey) [inline pyaedt/modules/SetupTemplates.py:2187]:
```suggestion
        with open(file_path, "w", encoding="utf-8") as f:
```

File: pyaedt/modules/SetupTemplates.py [hunk 6]
```diff
@@ -2172,3 +2173,40 @@ def __init__(self, setup, props):
...
+    def _export_properties_to_json(self, fpath):
+        """Export all setup properties into a json file.
+
+        Parameters
+        ----------
+        fpath : str
+            File path of the json file.
+        """
+        if not fpath.endswith(".json"):
+            fpath = fpath + ".json"
```
Comment (MaxJPRey) [inline pyaedt/modules/SetupTemplates.py:2186]:
```suggestion
    def _export_properties_to_json(self, file_path):
        """Export all setup properties into a json file.

        Parameters
        ----------
        file_path : str
            File path of the json file.
        """
        if not file_path.endswith(".json"):
            file_path = file_path + ".json"
```

File: pyaedt/modules/SetupTemplates.py (hunk 5 above)
Comment (MaxJPRey) [inline pyaedt/modules/SetupTemplates.py:2187]: @ring630 Do we want to override the file if it already exists? Is it ok? Or shall we raise an exception?

**Review by MaxJPRey — APPROVED** [review]
I mainly added some suggestions to replace the parameter and variables `fpath` by `file_path`.

File: pyaedt/modules/SetupTemplates.py (hunk 5 above)
Comment (ring630) [reply pyaedt/modules/SetupTemplates.py]: @MaxJPRey the user may or may not want to overwrite the existing file. We can add an argument overwrite=False

**Pull request merged**